teachertool/src/state/helpers.ts

```diff
@@ -1,7 +1,39 @@
-import { CatalogCriteria } from "../types/criteria";
+import { logError } from "../services/loggingService";
+import { CatalogCriteria, CriteriaInstance } from "../types/criteria";
+import { ErrorCode } from "../types/errorCode";
+import { Rubric } from "../types/rubric";
 import { stateAndDispatch } from "./appStateContext";
 
 export function getCatalogCriteriaWithId(id: string): CatalogCriteria | undefined {
     const { state } = stateAndDispatch();
     return state.catalog?.find(c => c.id === id);
 }
+
+export function verifyCriteriaInstanceIntegrity(instance: CriteriaInstance) {
```

Comment:
The function accepts a criteria instance and verifies that that instance's information (namely the catalog id and parameters) map to actual values in the catalog, so it's a bit of both, but from a caller's perspective, it's a scenario of "I have a criteria instance and I want to verify it's correctly defined."

I'm open to alternative name suggestions, but I did want to be careful to avoid possible confusion around verifying the instance's definition (which is what we're doing here) vs actually running validation of the code based on the criteria (hence weird "integrity" names).

Reply:
Oh, thanks for the clarification. The confusion I had was regarding testing the rubric integrity of an uploaded rubric. I was thinking that the rubric file wouldn't have instanced criteria, but that wouldn't make sense. Thanks!